File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-valuation-5-4/AssetValuation.cpp
// AssetValuation.cpp 
#include "AssetValuation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

AssetValuation::AssetValuation(TiXmlNode* xmlNode)
: Valuation(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //quoteNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* quoteNode = xmlNode->FirstChildElement("quote");

   if(quoteNode){quoteIsNull_ = false;}
   else{quoteIsNull_ = true;}

   if(quoteNode)
   {
      for(quoteNode; quoteNode; quoteNode = quoteNode->NextSiblingElement("quote")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- quoteNode , address : " << quoteNode << std::endl;
          #endif
          if(quoteNode->Attribute("href") || quoteNode->Attribute("id"))
          {
              if(quoteNode->Attribute("id"))
              {
                  quoteIDRef_ = quoteNode->Attribute("id");
                  quote_.push_back(boost::shared_ptr<Quotation>(new Quotation(quoteNode)));
                  quote_.back()->setID(quoteIDRef_);
                  IDManager::instance().setID(quoteIDRef_, quote_.back());
              }
              else if(quoteNode->Attribute("href")) { quoteIDRef_ = quoteNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { quote_.push_back(boost::shared_ptr<Quotation>(new Quotation(quoteNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- quoteNode , address : " << quoteNode << std::endl;
       #endif
   }

   //fxRateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* fxRateNode = xmlNode->FirstChildElement("fxRate");

   if(fxRateNode){fxRateIsNull_ = false;}
   else{fxRateIsNull_ = true;}

   if(fxRateNode)
   {
      for(fxRateNode; fxRateNode; fxRateNode = fxRateNode->NextSiblingElement("fxRate")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- fxRateNode , address : " << fxRateNode << std::endl;
          #endif
          if(fxRateNode->Attribute("href") || fxRateNode->Attribute("id"))
          {
              if(fxRateNode->Attribute("id"))
              {
                  fxRateIDRef_ = fxRateNode->Attribute("id");
                  fxRate_.push_back(boost::shared_ptr<FxRate>(new FxRate(fxRateNode)));
                  fxRate_.back()->setID(fxRateIDRef_);
                  IDManager::instance().setID(fxRateIDRef_, fxRate_.back());
              }
              else if(fxRateNode->Attribute("href")) { fxRateIDRef_ = fxRateNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { fxRate_.push_back(boost::shared_ptr<FxRate>(new FxRate(fxRateNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- fxRateNode , address : " << fxRateNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
std::vector<boost::shared_ptr<Quotation>> AssetValuation::getQuote()
{
   if(!this->quoteIsNull_){
        if(quoteIDRef_ != ""){
             return this->quote_;
        }else{
             return this->quote_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Quotation>>();
   }
}
std::vector<boost::shared_ptr<FxRate>> AssetValuation::getFxRate()
{
   if(!this->fxRateIsNull_){
        if(fxRateIDRef_ != ""){
             return this->fxRate_;
        }else{
             return this->fxRate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<FxRate>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-valuation-5-4/AssetValuation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class AssetValuation : Valuation
    {
        public AssetValuation(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNodeList quoteNodeList = xmlNode.SelectNodes("quote");
            
            if (quoteNodeList != null)
            {
                this.quote_ = new List<Quotation>();
                foreach (XmlNode item in quoteNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            quoteIDRef_ = item.Attributes["id"].Value;
                            quote_.Add(new Quotation(item));
                            IDManager.SetID(quoteIDRef_, quote_[quote_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            quoteIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        quote_.Add(new Quotation(item));
                        }
                    }
                    else
                    {
                        quote_.Add(new Quotation(item));
                    }
                }
            }
            
        
            XmlNodeList fxRateNodeList = xmlNode.SelectNodes("fxRate");
            
            if (fxRateNodeList != null)
            {
                this.fxRate_ = new List<FxRate>();
                foreach (XmlNode item in fxRateNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            fxRateIDRef_ = item.Attributes["id"].Value;
                            fxRate_.Add(new FxRate(item));
                            IDManager.SetID(fxRateIDRef_, fxRate_[fxRate_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            fxRateIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        fxRate_.Add(new FxRate(item));
                        }
                    }
                    else
                    {
                        fxRate_.Add(new FxRate(item));
                    }
                }
            }
            
        
        }
        
    
        #region Quote_
        private List<Quotation> quote_;
        public List<Quotation> Quote_
        {
            get
            {
                if (this.quote_ != null)
                {
                    return this.quote_; 
                }
                else if (this.quoteIDRef_ != null)
                {
                    return this.quote_; 
                }
                else
                {
                      return this.quote_; 
                }
            }
            set
            {
                if (this.quote_ != value)
                {
                    this.quote_ = value;
                }
            }
        }
        #endregion
        
        public string quoteIDRef_ { get; set; }
        #region FxRate_
        private List<FxRate> fxRate_;
        public List<FxRate> FxRate_
        {
            get
            {
                if (this.fxRate_ != null)
                {
                    return this.fxRate_; 
                }
                else if (this.fxRateIDRef_ != null)
                {
                    return this.fxRate_; 
                }
                else
                {
                      return this.fxRate_; 
                }
            }
            set
            {
                if (this.fxRate_ != value)
                {
                    this.fxRate_ = value;
                }
            }
        }
        #endregion
        
        public string fxRateIDRef_ { get; set; }
        
    
        
    
    }
    
}

